ds3-autogen-cli/src/main/java/com/spectralogic/autogen/cli/Arguments.java

```diff
@@ -2,18 +2,29 @@
 
 
 public class Arguments {
-    private final String targetDir;
-    private final GeneratorType type;
-    private final String inputSpec;
-    private final boolean help;
+    private String targetDir;
+    private GeneratorType type;
+    private String inputSpec;
+    private boolean help;
 
-    public Arguments(final String targetDir, final GeneratorType type, final String inputSpec, final boolean help) {
```

Comment:
The variables were removed from the constructor parameters in order to prevent the necessity of all values existing.  This was what caused the need for all params in order to run the help message.  Since they are no longer initialized in the constructor, they cannot be final.